streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/CheckpointProcessor.java

```diff
@@ -108,6 +112,23 @@ public void process(Long appId, CheckPoints checkPoints) {
     }
   }
 
+  private boolean shouldProcessFailedTrigger(
+      CheckPoints.CheckPoint checkPoint, Application application, CheckPointStatus status) {
+    return CheckPointStatus.FAILED.equals(status)
+        && !checkPoint.getIsSavepoint()
+        && application.cpFailedTrigger();
+  }
+
+  private boolean checkpointNeedStore(
+      Application application, @Nonnull CheckPoints.CheckPoint checkPoint) {
+    LambdaQueryWrapper<SavePoint> queryWrapper =
```

Comment:
We should check the `checkPointCache` first, right?

Reply:
A minor confusion for me form the comments.
Does the minor refactor cause the logic loss or the original implementation?

Reply:
did you mean we should judge if the chk/savepoint id exist in the cache?

Reply:
I mean we still use the old logic.

The code of the master branch determines whether checkpoint/savepoint needs to be stored in the database according to the cache, and I think it is enough for now. Your code will access to the database frequently, it isn't friendly to database.